sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -129,6 +129,7 @@ static void handleVisitedNodes(std::vector<Command *> &Visited) {
   for (Command *Cmd : Visited) {
     if (Cmd->MMarks.MToBeDeleted) {
       Cmd->getEvent()->setCommand(nullptr);
+      Cmd->getEvent()->cleanupDependencyEvents();
```

Comment:
This seems to still lead to d-tor to go through DFS and overflowing the stack, isn't it?

Reply:
Last changes are not lead to stack overflow on the tests that were affected